### PR TITLE
Add pivot table warning if its wrapper has no or small height

### PR DESCRIPTION
**Pull request by Tulean**
JIRA: RAIL-2055

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-ui-pivot/package.json [hunk 1]
```diff
@@ -122,6 +123,7 @@
         "stylelint-config-prettier": "^8.0.2",
         "stylelint": "^13.8.0",
         "ts-jest": "^26.3.0",
-        "typescript": "4.0.2"
+        "typescript": "4.0.2",
+        "@types/uuid": "^8.3.0"
```
Comment (no23reason) [inline libs/sdk-ui-pivot/package.json:127]: Please keep the alphabetical order.

File: libs/sdk-ui-pivot/src/CorePivotTable.tsx [hunk 2]
```diff
@@ -352,6 +357,23 @@ export class CorePivotTableAgImpl extends React.Component<ICorePivotTableProps,
...
+     * on what is happening.
+     *
+     * @private
+     */
+    private alertParentWrapperMissingHeight(): void {
+        const parentWrapper = document.getElementById(this.pivotTableId)?.parentElement;
+        const parentHeight = parentWrapper?.style?.height || "0";
+        if (isEmpty(parentHeight) || parseInt(parentHeight) < 20) {
+            console.warn(
+                `The wrapper height of the pivot table has not been set or is suspiciously small. This might cause pivot table rendering issues. If so, please set an appropriate height for the wrapper. Use "${this.pivotTableId}" to find the PivotTable element in the DOM, which will help you to identify its wrapper.`,
```
Comment (no23reason) [inline libs/sdk-ui-pivot/src/CorePivotTable.tsx:372]: Please provide an example of how to do it (document.getElementById for example).

File: libs/sdk-ui-pivot/src/CorePivotTable.tsx [hunk 3]
```diff
@@ -352,6 +357,23 @@ export class CorePivotTableAgImpl extends React.Component<ICorePivotTableProps,
...
+     * Checks DOM if height has been set on PivotTable parent wrapper. If height has not been set,
+     * PivotTable will not be rendered correctly. Therefore, we need to inform the user at least on the console
+     * on what is happening.
+     *
+     * @private
+     */
+    private alertParentWrapperMissingHeight(): void {
+        const parentWrapper = document.getElementById(this.pivotTableId)?.parentElement;
+        const parentHeight = parentWrapper?.style?.height || "0";
+        if (isEmpty(parentHeight) || parseInt(parentHeight) < 20) {
```
Comment (no23reason) [inline libs/sdk-ui-pivot/src/CorePivotTable.tsx:370]: This will trigger false positives for for example "19em", no?

Comment (no23reason) [reply libs/sdk-ui-pivot/src/CorePivotTable.tsx:370]: Also, thanks to the `|| "0"` the `parentHeight` will never be empty.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3224/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4880/ : FAILURE in 6m 15s

File: libs/sdk-ui-pivot/src/CorePivotTable.tsx (hunk 3 above)
Comment (Tulean) [reply libs/sdk-ui-pivot/src/CorePivotTable.tsx]: Good catch, unfortunately we don't differentiate em, px, %.... Instead I replaced checking styling by checking the element height itself.

Comment (no23reason) [reply libs/sdk-ui-pivot/src/CorePivotTable.tsx]: That is better :)

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-pivot/src/CorePivotTable.tsx [hunk 4]
```diff
@@ -352,6 +356,24 @@ export class CorePivotTableAgImpl extends React.Component<ICorePivotTableProps,
...
+    /**
+     * Checks DOM if height has been set on PivotTable parent wrapper. If height has not been set,
+     * PivotTable will not be rendered correctly. Therefore, we need to inform the user at least on the console
+     * on what is happening.
+     *
+     * @private
+     */
+    private alertParentWrapperMissingHeight(): void {
+        const parentWrapper = document.getElementById(this.pivotTableId)?.parentElement;
+        const parentHeight = parentWrapper?.offsetHeight || "0";
```
Comment (no23reason) [inline libs/sdk-ui-pivot/src/CorePivotTable.tsx:368]: Now that parentHeight is number, this should be `const parentHeight = parentWrapper?.offsetHeight ?? 0;`

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1257

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/970/ : SUCCESS in 3m 10s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1240/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4888/ : FAILURE in 16m 30s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/356/ : SUCCESS in 11m 24s (non-voting)

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1257

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/971/ : SUCCESS in 2m 45s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1241/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/4892/ : SUCCESS in 12m 50s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-el7/357/ : SUCCESS in 8m 51s (non-voting)

**Pull request merged**